server/src/handler/mod.rs

```diff
@@ -16,6 +19,18 @@ impl<Deps: TpmContextDeps> CommandHandler<Deps> {
     pub fn new(crypto: Deps::Crypto) -> Self {
         Self { crypto }
     }
+    /// invariant: buffer will fit into TPM2B_DIGEST which is much less
+    /// than the maximum request of SP 800 90A
+    fn infallable_get_random(&mut self, buffer: &mut [u8]) {
```

Comment:
Like was mentioned by Brad, I agree with the idea of these "should never fail" functions triggering failure mode. Ideally we'd set this up and make the ability to trigger this mode a dependency of doing this change, but as long as you leave a TODO in comments, that'd be workable. It basically points to a bug in the server implementation.

Reply:
yes, I am working on fixing that, but I am afraid that will make the patch too big so I am putting a place holder `todo!() // we go to failure mode` I hope that is ok for the time being!